okhttp-loom/src/main/kotlin/okhttp3/loom/LoomBackend.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2022 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3.loom
+
+import java.util.concurrent.BlockingQueue
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Executors
+import okhttp3.internal.concurrent.TaskRunner
+import okhttp3.internal.notify
+
+/**
+ * May not be needed, if doesn't change from real backend.
+ */
+class LoomBackend(
+  internal val executor: ExecutorService = Executors.newVirtualThreadPerTaskExecutor()
+) : TaskRunner.Backend {
+  override fun nanoTime(): Long = System.nanoTime()
+
+
+  override fun coordinatorNotify(taskRunner: TaskRunner) {
+    taskRunner.notify()
```

Comment:
@swankjesse whats the best update for these synchronization primitives?

Reply:
Do the Loom authors offer any guidance? I think we want a Lock or a Monitor.

Reply:
Taking a look at Conditions. https://stackoverflow.com/a/10407588